addons/web_editor/static/src/js/wysiwyg/widgets/link.js

```diff
@@ -1,47 +1,83 @@
 /** @odoo-module alias=wysiwyg.widgets.Link **/
 
-import core from "web.core";
 import * as OdooEditorLib from "@web_editor/js/editor/odoo-editor/src/OdooEditor";
-import Widget from "web.Widget";
 import {isColorGradient} from "web_editor.utils";
+import {
+    Component,
+    onWillStart,
+    onMounted,
+    onWillUpdateProps,
+    onWillDestroy,
+    useState,
+    useRef,
+} from "@odoo/owl";
 
 const getDeepRange = OdooEditorLib.getDeepRange;
 const getInSelection = OdooEditorLib.getInSelection;
-const _t = core._t;
+import { _t } from "@web/core/l10n/translation";
 
 /**
  * Allows to customize link content and style.
  */
-const Link = Widget.extend({
-    events: {
-        'input': '_onAnyChange',
-        'change': '_onAnyChange',
-        'input input[name="url"]': '__onURLInput',
-        'change input[name="url"]': '_onURLInputChange',
-    },
+export class Link extends Component {
+    static props = {
+        editable: true,
+        link: true,
+        needLabel: { type: Boolean, optional: true },
+        forceNewWindow: { type: Boolean, optional: true },
+        initialIsNewWindow: { type: Boolean, optional: true },
+        shouldFocusUrl: { type: Boolean, optional: true },
+    };
+    static defaultProps = {
+        needLabel: true,
+        forceNewWindow: false,
+        initialIsNewWindow: false,
+        shouldFocusUrl: false,
+    }
+    linkComponentWrapperRef = useRef("linkComponentWrapper");
+    setup() {
+        this.state = useState({});
+
+        onWillStart(() => this.onWillStart());
+        let started = false;
+        onMounted(async () => {
+            if (started) {
+                return;
+            }
+            started = true;
+            // Todo: at the time this method is called, the ref is not yet in
+            // the dom and this.linkComponentWrapperRef.el returns null as the
+            // getter check the node with `inOwnerDocument`. This hack allows to
+            // get the node without the check.
+            this.$el = $(this.__owl__.refs.linkComponentWrapper);
 
-    /**
-     * @constructor
-     * @param {Boolean} data.isButton - whether if the target is a button element.
-     */
-    init: function (parent, options, editable, data, $button, link) {
-        this.options = options || {};
-        this._super(parent, Object.assign({
-            title: _t("Link to"),
-        }, this.options));
+            this.$ = this.$el.find.bind(this.$el);
 
-        this.data = data || {};
-        this.isButton = this.data.isButton;
-        this.$button = $button;
-        this.noFocusUrl = this.options.noFocusUrl;
+            this.$el.find('input, select').on('input', this._onAnyChange.bind(this));
+            this.$el.find('input, select').on('change', this._onAnyChange.bind(this));
+            this.$el.find('[name="url"]').on('input', this.__onURLInput.bind(this));
+            this.$el.find('[name="url"]').on('change', this._onURLInputChange.bind(this));
```

Comment:
This does not seem very owl-like and also if something is done in onMounted, should not it be undone in onUnmounted or something like that?